Update CONTRIBUTING.md and related files

wanyamaman opened the pull request:
- Explicitly lists the locations to run commands when setting up the gem for local development.
- Fixes an issue where `yarn install-react-on-rails` command used `yarn` instead of `yarn add` and failed to exectute.

Closes #736

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/shakacode/react_on_rails/737)
<!-- Reviewable:end -->

justin808 commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[CONTRIBUTING.md, line 74 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhW2zeelnvA9aKbVQn:-KdhW2zeelnvA9aKbVQo:b-dugd3o) ([raw file](https://github.com/shakacode/react_on_rails/blob/9a198bf716d74f2fbf0a62c43a23f1d377bc4fa2/CONTRIBUTING.md#L74)):*
> ```Markdown
> * /spec/react_on_rails/dummy-for-generators
> 
> # Configuring your test app to use your local fork
> ```

A test app could be a brand new rails app configured with the React on Rails gem.

---

*[CONTRIBUTING.md, line 100 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhVpLhPz9DwTuk60Kp:-KdhVpLhPz9DwTuk60Kq:b-bz56kv) ([raw file](https://github.com/shakacode/react_on_rails/blob/9a198bf716d74f2fbf0a62c43a23f1d377bc4fa2/CONTRIBUTING.md#L100)):*
> ```Markdown
> 
> When you use a relative path, be sure to run the above `yarn` command whenever you change the node package for react-on-rails.
> ```

Why is all of this being removed?

Unless there is a specific reason for removing one of these steps, please keep them in.

OK -- I see, you believe this should go below?

---

*[CONTRIBUTING.md, line 152 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWtpFC7qOerdYZJSE:-KdhWtpFC7qOerdYZJSF:b-aturyh) ([raw file](https://github.com/shakacode/react_on_rails/blob/81d2fd6ff1f5378bf6709cb8e2fadec1372f457c/CONTRIBUTING.md#L152)):*
> ```Markdown
> 
> ```sh
> # react_on_rails/spec/dummy/client
> ```

maybe just make it clear that this is directory to cd to?

---

*[CONTRIBUTING.md, line 164 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhW8ykmm_hG406Yz8_:-KdhW8ykmm_hG406Yz8a:b-57pia0) ([raw file](https://github.com/shakacode/react_on_rails/blob/81d2fd6ff1f5378bf6709cb8e2fadec1372f457c/CONTRIBUTING.md#L164)):*
> ```Markdown
> 
> ### Starting the Dummy App
> To run the dummy app, it's **CRITICAL** to not just run `rails s`. You have to run `foreman start`. If you don't do this, then `webpack` will not generate a new bundle, and you will be seriously confused when you change JavaScript and the app does not change. If you change the webpack configs, then you need to restart foreman. If you change the JS code for react-on-rails, you need to run `yarn run build`. Since the react-on-rails package should be sym linked, you don't have to `yarn react-on-rails` after every change.
> ```

The `spec/dummy` is a special "test app" that lives inside of the gem and is used for some tests. We also have direct gem tests and test of the generator.

---

*[CONTRIBUTING.md, line 170 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWo5SIUz7ERZyU40_:-KdhWo5SIUz7ERZyU40a:b9ejmh9) ([raw file](https://github.com/shakacode/react_on_rails/blob/9a198bf716d74f2fbf0a62c43a23f1d377bc4fa2/CONTRIBUTING.md#L170)):*
> ```Markdown
> 
> ```sh
> cd spec/dummy
> ```

yes, you need to cd here to run these tests

---

*[CONTRIBUTING.md, line 177 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWr56pY7TUxNMCX6W:-KdhWr56pY7TUxNMCX6X:bmf440r) ([raw file](https://github.com/shakacode/react_on_rails/blob/9a198bf716d74f2fbf0a62c43a23f1d377bc4fa2/CONTRIBUTING.md#L177)):*
> ```Markdown
> 
> ```sh
> cd spec/dummy/client
> ```

yes, you need to be here

---

*[CONTRIBUTING.md, line 220 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWM5zsKU7UKwzVCj5:-KdhWM5zsKU7UKwzVCj6:b-swu2cv) ([raw file](https://github.com/shakacode/react_on_rails/blob/81d2fd6ff1f5378bf6709cb8e2fadec1372f457c/CONTRIBUTING.md#L220)):*
> ```Markdown
> Run `docker-compose build tests` to build the tests container. Run `docker-compose run tests` to start all RSpec tests.
> 
> # Configuring your test app to use your local fork
> ```

I guess we can make it clear what's are the different types of test apps.

---

*[spec/dummy/client/package.json, line 77 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWT6DvaA42qpgZ0Fd:-KdhWT6DvaA42qpgZ0Fe:b-42mgss) ([raw file](https://github.com/shakacode/react_on_rails/blob/81d2fd6ff1f5378bf6709cb8e2fadec1372f457c/spec/dummy/client/package.json#L77)):*
> ```JSON
>     "build:server": "webpack --config webpack.server.rails.build.config.js",
>     "hot-assets": "babel-node server-rails-hot.js",
>     "install-react-on-rails": "yarn add 'file:../../..'"
> ```

this looks good if you tried it!

and please be sure that when you change the JS part of the npm package, that just running this command works...

I'm not 100% sure that this runs the build process on the package. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737)*
<!-- Sent from Reviewable.io -->

wanyamaman commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[CONTRIBUTING.md, line 100 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhVpLhPz9DwTuk60Kp:-KdhZjNd8Wc87lBO6lZ5:bkgra9t) ([raw file](https://github.com/shakacode/react_on_rails/blob/9a198bf716d74f2fbf0a62c43a23f1d377bc4fa2/CONTRIBUTING.md#L100)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Why is all of this being removed?

Unless there is a specific reason for removing one of these steps, please keep them in.

OK -- I see, you believe this should go below?
</blockquote></details>

I'm not entirely comfortable with it being all the way at the bottom too. But it did confuse me by the time I reached the section on _Development Setup for Gem and Node Package Contributors_. You'd have been introduced to the test app outside the gem, the gem root itself as well as the dummy app. That's three different places you can run `yarn` commands.

Alternatively it can stay in its original place but we must be clear on what is a *test* app vs a dummy app vs example app and where commands are to be run. 

---

*[CONTRIBUTING.md, line 164 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhW8ykmm_hG406Yz8_:-KdhcORjnMSaH9RCQ7vO:by0l9fa) ([raw file](https://github.com/shakacode/react_on_rails/blob/81d2fd6ff1f5378bf6709cb8e2fadec1372f457c/CONTRIBUTING.md#L164)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

The `spec/dummy` is a special "test app" that lives inside of the gem and is used for some tests. We also have direct gem tests and test of the generator.
</blockquote></details>

Thanks for the confirmation. I managed to figure it out eventually. However, it does get confusing if we sometimes call it the dummy app and at other places the test app since we have test apps that live outside the gem as well. Do you think we should be able to distinguish between the two?

---

*[CONTRIBUTING.md, line 220 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWM5zsKU7UKwzVCj5:-KdhdhPj46njb5yz4zb7:bjo5g6w) ([raw file](https://github.com/shakacode/react_on_rails/blob/81d2fd6ff1f5378bf6709cb8e2fadec1372f457c/CONTRIBUTING.md#L220)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

I guess we can make it clear what's are the different types of test apps.
</blockquote></details>

Yes. This is where most of the confusion stems from.

---

*[spec/dummy/client/package.json, line 77 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWT6DvaA42qpgZ0Fd:-KdheepLqDMliBPFMQaj:bu0v5yz) ([raw file](https://github.com/shakacode/react_on_rails/blob/81d2fd6ff1f5378bf6709cb8e2fadec1372f457c/spec/dummy/client/package.json#L77)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

this looks good if you tried it!

and please be sure that when you change the JS part of the npm package, that just running this command works...

I'm not 100% sure that this runs the build process on the package. 
</blockquote></details>

Gotcha. I'll confirm the changes. Just to be sure, we want this to be the equivalent of running `yarn` and `yarn run build` at the root directory of the gem. Is that correct?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[CONTRIBUTING.md, line 100 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhVpLhPz9DwTuk60Kp:-KdiezVsm6kg211mZERK:bm9c8gm) ([raw file](https://github.com/shakacode/react_on_rails/blob/9a198bf716d74f2fbf0a62c43a23f1d377bc4fa2/CONTRIBUTING.md#L100)):*
<details><summary><i>Previously, wanyamaman (William Wanyama) wrote…</i></summary><blockquote>

I'm not entirely comfortable with it being all the way at the bottom too. But it did confuse me by the time I reached the section on _Development Setup for Gem and Node Package Contributors_. You'd have been introduced to the test app outside the gem, the gem root itself as well as the dummy app. That's three different places you can run `yarn` commands.

Alternatively it can stay in its original place but we must be clear on what is a *test* app vs a dummy app vs example app and where commands are to be run. 
</blockquote></details>

I think up at the top is good as this setup is used for /spec/dummy.

Let's make it very clear that the /spec/dummy is one test case for react on rails, and an example of the various techniques!

---

*[CONTRIBUTING.md, line 164 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhW8ykmm_hG406Yz8_:-KdifL_Q7ZLyZLLmjewE:byyja1y) ([raw file](https://github.com/shakacode/react_on_rails/blob/81d2fd6ff1f5378bf6709cb8e2fadec1372f457c/CONTRIBUTING.md#L164)):*
<details><summary><i>Previously, wanyamaman (William Wanyama) wrote…</i></summary><blockquote>

Thanks for the confirmation. I managed to figure it out eventually. However, it does get confusing if we sometimes call it the dummy app and at other places the test app since we have test apps that live outside the gem as well. Do you think we should be able to distinguish between the two?
</blockquote></details>

we can just call it the `spec/dummy` app

---

*[CONTRIBUTING.md, line 170 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWo5SIUz7ERZyU40_:-KdifEZ_7jCgPrfHJ9bl:b-gox0r1) ([raw file](https://github.com/shakacode/react_on_rails/blob/9a198bf716d74f2fbf0a62c43a23f1d377bc4fa2/CONTRIBUTING.md#L170)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

yes, you need to cd here to run these tests
</blockquote></details>

Please make it clear you have to be in these directories.

---

*[spec/dummy/client/package.json, line 77 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWT6DvaA42qpgZ0Fd:-KdifV1wIOPzd1z-YOlH:b4grlxu) ([raw file](https://github.com/shakacode/react_on_rails/blob/81d2fd6ff1f5378bf6709cb8e2fadec1372f457c/spec/dummy/client/package.json#L77)):*
<details><summary><i>Previously, wanyamaman (William Wanyama) wrote…</i></summary><blockquote>

Gotcha. I'll confirm the changes. Just to be sure, we want this to be the equivalent of running `yarn` and `yarn run build` at the root directory of the gem. Is that correct?
</blockquote></details>

test it out...try it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737)*
<!-- Sent from Reviewable.io -->

justin808 commented:
a few comments...

---

Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-:-KdifYDgVVc3h69I6bs9:b-e0s4aa)*
<!-- Sent from Reviewable.io -->

justin808 commented:
@wanyamaman You've got some odd things going on with your git commits. I want to see a PR with ONE commit with your changes.

justin808 commented:
This is looking **FABULOUS**. Let's just clarify so that the noob knows that there's 2 parts for the local development reference: 1) ruby part, 2) node part.

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[CONTRIBUTING.md, line 82 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxGhV7zgJ9vaiiketb:-KdxGhV7zgJ9vaiiketc:bd8o826) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L82)):*
> ```Markdown
>     └── spec
>         └── dummy
> ```
> ```

nifty graphic!

---

*[CONTRIBUTING.md, line 85 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxGoxdD1cpXflQfyGO:-KdxGoxdD1cpXflQfyGP:b642n47) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L85)):*
>  to test the gem 

`to test the ruby parts of the gem`

and let's be clear on needing to run either the ruby part or the the NPM part depending on if you want to test the ruby or npm parts or **BOTH**.

---

*[CONTRIBUTING.md, line 105 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxH8sl8PmXhk-PPBqp:-KdxH8sl8PmXhk-PPBqq:b-yz0g23) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L105)):*
> ```Markdown
> ```sh
> cd test/client
> yarn add "react-on-rails@../../react_on_rails"
> ```

did you try this?

---

*[CONTRIBUTING.md, line 108 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxHBVj9VwTTC6AWVCt:-KdxHBVj9VwTTC6AWVCu:b666fb5) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L108)):*
> ```Markdown
> ```
> 
> When you use a relative path, be sure to run the above `yarn` command whenever you change the node package for react-on-rails.
> ```

and not just running yarn install as that might not do anything unless the underlying package.json version number changes.

---

*[spec/dummy/client/package.json, line 77 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxHX1JKN-_eD-7iXmo:-KdxHX1JKN-_eD-7iXmp:bpw7vqr) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/spec/dummy/client/package.json#L77)):*
> ```JSON
>     "build:server": "webpack --config webpack.server.rails.build.config.js",
>     "hot-assets": "babel-node server-rails-hot.js",
>     "install-react-on-rails": "CLIENT_PATH=$(pwd) && cd ../../../ && yarn && cd $CLIENT_PATH && yarn add 'file:../../..'"
> ```

Is the use of the CLIENT_PATH variable required? Nifty if this what it took.

we should probably assume that the script  `yarn run install-react-on-rails` is run from the same directory...

anyway if this is required, super....

if not, then let's remove.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-:-KdxI3tOuFiQfkkCjdhC:b-qu57o7)*
<!-- Sent from Reviewable.io -->

wanyamaman commented:
Review status: all files reviewed at latest revision, 13 unresolved discussions, some commit checks failed.

---

*[CONTRIBUTING.md, line 74 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhW2zeelnvA9aKbVQn:-Ke-jpSmxqItH33AGeCv:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/9a198bf716d74f2fbf0a62c43a23f1d377bc4fa2/CONTRIBUTING.md#L74)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

A test app could be a brand new rails app configured with the React on Rails gem.
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 100 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhVpLhPz9DwTuk60Kp:-Ke-k93UQ9TsuQDf6HP2:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/9a198bf716d74f2fbf0a62c43a23f1d377bc4fa2/CONTRIBUTING.md#L100)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

I think up at the top is good as this setup is used for /spec/dummy.

Let's make it very clear that the /spec/dummy is one test case for react on rails, and an example of the various techniques!
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 152 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWtpFC7qOerdYZJSE:-Ke-kElgKI0AkKVej5sh:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/81d2fd6ff1f5378bf6709cb8e2fadec1372f457c/CONTRIBUTING.md#L152)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

maybe just make it clear that this is directory to cd to?
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 170 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWo5SIUz7ERZyU40_:-Ke-lho34tspJ9Owev07:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/9a198bf716d74f2fbf0a62c43a23f1d377bc4fa2/CONTRIBUTING.md#L170)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Please make it clear you have to be in these directories.
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 177 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWr56pY7TUxNMCX6W:-Ke-kDlWsvlgiO39Z1sn:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/9a198bf716d74f2fbf0a62c43a23f1d377bc4fa2/CONTRIBUTING.md#L177)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

yes, you need to be here
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 85 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxGoxdD1cpXflQfyGO:-Ke08tEtg0fnFCTBdIJi:b-ytrqyi) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L85)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

>  to test the gem 

`to test the ruby parts of the gem`

and let's be clear on needing to run either the ruby part or the the NPM part depending on if you want to test the ruby or npm parts or **BOTH**.

</blockquote></details>

 I'm not sure what the overall output of this section is meant to be. Please clarify

---

*[CONTRIBUTING.md, line 93 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-Ke08ydNCsuPX9j-s7G9:-Ke08ydNCsuPX9j-s7GA:b4pvx5x) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L93)):*
> ```Markdown
> Note that you will need to bundle install after making this change, but also that **you will need to restart your Rails application if you make any changes to the gem**.
> 
> ## NPM for react-on-rails
> ```

This section needs more information. Is this the part when the user wants to test the npm part of the gem. Do we assume they've already installed gem or is this a continuous sequence from the Ruby gem section above?

---

*[CONTRIBUTING.md, line 105 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxH8sl8PmXhk-PPBqp:-Ke090P0z1JIHVO0_Lyv:bxcit6i) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L105)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

did you try this?
</blockquote></details>

Yes, this adds the following line to the `client/package.json` dependencies ->
`"react-on-rails": "../../react_on_rails",`

---

*[CONTRIBUTING.md, line 108 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxHBVj9VwTTC6AWVCt:-Ke-loskAvy5gdj78dyN:b-5po8e5) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L108)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

and not just running yarn install as that might not do anything unless the underlying package.json version number changes.
</blockquote></details>

doesn't the command `yarn` call `yarn install` if you don't pass it any commands?

---

*[spec/dummy/client/package.json, line 77 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxHX1JKN-_eD-7iXmo:-Ke-kMb-l6xnLlEeWetB:b-ia7vsd) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/spec/dummy/client/package.json#L77)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Is the use of the CLIENT_PATH variable required? Nifty if this what it took.

we should probably assume that the script  `yarn run install-react-on-rails` is run from the same directory...

anyway if this is required, super....

if not, then let's remove.
</blockquote></details>

It depends. If we will have other dummy apps besides the spec dummy then this is a universal script that will work with them all. Otherwise we can hard code the path down from the gem root to the `spec/dummy` -> which would be:
`cd ../../../ && yarn && cd spec/dummy/client && yarn add 'file:../../../'`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks broke.

---

*[CONTRIBUTING.md, line 85 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxGoxdD1cpXflQfyGO:-Ke1e_H0kX1XFHwgleLN:b-1md686) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L85)):*
<details><summary><i>Previously, wanyamaman (William Wanyama) wrote…</i></summary><blockquote>

 I'm not sure what the overall output of this section is meant to be. Please clarify
</blockquote></details>

Hi @wanyamaman, the idea is that you will change either the JS part or the Ruby part or both. Depending on which you change, you need  to link to the relevant part. One other detail is that the Ruby  part just requires restarting the server. The JavaScript part requires re-running the install script and refreshing the browser page.

---

*[CONTRIBUTING.md, line 93 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-Ke08ydNCsuPX9j-s7G9:-Ke1clU1Q5hE0-2f8T4I:b-fz7gxb) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L93)):*
Hi @wanyamaman! Yes! 

> Is this the part when the user wants to test the npm part of the gem

---

*[CONTRIBUTING.md, line 108 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxHBVj9VwTTC6AWVCt:-Ke1dDgpP-jAfwwsnqD6:b-ufijkz) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L108)):*
<details><summary><i>Previously, wanyamaman (William Wanyama) wrote…</i></summary><blockquote>

doesn't the command `yarn` call `yarn install` if you don't pass it any commands?
</blockquote></details>

Yes, yarn does do `yarn install` by default. Older versions of npm **DID NOT** update the JS files under node_modules for a locally referenced directory each time. This may have changed with yarn. The way to check is to make a JS file change in this directory: https://github.com/shakacode/react_on_rails/tree/master/node_package/src and to then run the `yarn` command with no parameters and to see if the file under `/client/node_modules` in your test app changed.

---

*[spec/dummy/client/package.json, line 77 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxHX1JKN-_eD-7iXmo:-Ke1e1-GT5hL5nN4MQuu:brq61uu) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/spec/dummy/client/package.json#L77)):*
<details><summary><i>Previously, wanyamaman (William Wanyama) wrote…</i></summary><blockquote>

It depends. If we will have other dummy apps besides the spec dummy then this is a universal script that will work with them all. Otherwise we can hard code the path down from the gem root to the `spec/dummy` -> which would be:
`cd ../../../ && yarn && cd spec/dummy/client && yarn add 'file:../../../'`
</blockquote></details>

I think we should assume this command is specific to this directory. I'm guessing that the cd to the top level directory is needed to build the node package?

I'm surprised we cannot do something like just running the command that you mention above:

`yarn add "react-on-rails@../../../react_on_rails"`

(number of directories up might not be right)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737)*
<!-- Sent from Reviewable.io -->

wanyamaman commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[CONTRIBUTING.md, line 164 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhW8ykmm_hG406Yz8_:-Ke6etCwKMA6LUR-CUcE:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/81d2fd6ff1f5378bf6709cb8e2fadec1372f457c/CONTRIBUTING.md#L164)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

we can just call it the `spec/dummy` app
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 108 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxHBVj9VwTTC6AWVCt:-Ke6ouB268W8qikOG7x6:bqgq1su) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L108)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Yes, yarn does do `yarn install` by default. Older versions of npm **DID NOT** update the JS files under node_modules for a locally referenced directory each time. This may have changed with yarn. The way to check is to make a JS file change in this directory: https://github.com/shakacode/react_on_rails/tree/master/node_package/src and to then run the `yarn` command with no parameters and to see if the file under `/client/node_modules` in your test app changed.
</blockquote></details>

Hmm I'm not getting the file in client/modules to change. Thoughts?

---

*[spec/dummy/client/package.json, line 77 at r1](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdhWT6DvaA42qpgZ0Fd:-Ke6eb46iRR06wxISig5:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/81d2fd6ff1f5378bf6709cb8e2fadec1372f457c/spec/dummy/client/package.json#L77)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

test it out...try it.
</blockquote></details>

Done.

---

*[spec/dummy/client/package.json, line 77 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxHX1JKN-_eD-7iXmo:-Ke6YB7ggW5avroWW3r1:b3vve40) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/spec/dummy/client/package.json#L77)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

I think we should assume this command is specific to this directory. I'm guessing that the cd to the top level directory is needed to build the node package?

I'm surprised we cannot do something like just running the command that you mention above:

`yarn add "react-on-rails@../../../react_on_rails"`

(number of directories up might not be right)
</blockquote></details>

Unfortunately, using yarn add alone isn't triggering the build process for the gem but it does install it in dummy apps dependency list.

I'll change the command to be directory specific.

_The number of directories is fine but lets change it to be more explicit when reading_

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[CONTRIBUTING.md, line 108 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxHBVj9VwTTC6AWVCt:-Ke7ETz5MRiDRI4A6jc3:b-a1lnw) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L108)):*
<details><summary><i>Previously, wanyamaman (William Wanyama) wrote…</i></summary><blockquote>

Hmm I'm not getting the file in client/modules to change. Thoughts?
</blockquote></details>

this is what needs to be clear. 

I usually use a command like:

https://github.com/shakacode/react_on_rails/blob/master/spec/dummy/package.json#L14

```
    "install-react-on-rails": "cd client && yarn run install-react-on-rails",
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737)*
<!-- Sent from Reviewable.io -->

wanyamaman commented:
Review status: 0 of 2 files reviewed at latest revision, 6 unresolved discussions.

---

*[CONTRIBUTING.md, line 82 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxGhV7zgJ9vaiiketb:-KeFdL5JAftRIv7ghRe0:b-k73z76) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L82)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

nifty graphic!
</blockquote></details>

thanks

---

*[CONTRIBUTING.md, line 85 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxGoxdD1cpXflQfyGO:-KeFdQtVBlrHZHaFU309:b3nozzs) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L85)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Hi @wanyamaman, the idea is that you will change either the JS part or the Ruby part or both. Depending on which you change, you need  to link to the relevant part. One other detail is that the Ruby  part just requires restarting the server. The JavaScript part requires re-running the install script and refreshing the browser page.
</blockquote></details>

I've added clarification.

---

*[CONTRIBUTING.md, line 108 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KdxHBVj9VwTTC6AWVCt:-KeFdo9YnJ5Bo1EjjTI_:b-zbofx5) ([raw file](https://github.com/shakacode/react_on_rails/blob/80f4f14b2d7e7094f294431c243b3bbbffaacae4/CONTRIBUTING.md#L108)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

this is what needs to be clear. 

I usually use a command like:

https://github.com/shakacode/react_on_rails/blob/master/spec/dummy/package.json#L14

```
    "install-react-on-rails": "cd client && yarn run install-react-on-rails",
```


</blockquote></details>

Reverted back to npm.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737)*
<!-- Sent from Reviewable.io -->

justin808 commented:
some comments

getting closer!

---

Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks broke.

---

*[CONTRIBUTING.md, line 85 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIB7Jjdjj0LM7_z6n1:-KeIB7Jjdjj0LM7_z6n2:b-qdtenn) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/CONTRIBUTING.md#L85)):*
> ```Markdown
> ```
> 
> ## Test Ruby Gem
> ```

Testing the Ruby Gem

---

*[CONTRIBUTING.md, line 94 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIBBO7X70btnvBYnV8:-KeIBBO7X70btnvBYnV9:bjehljg) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/CONTRIBUTING.md#L94)):*
> ```Markdown
> Note that you will need to bundle install after making this change, but also that **you will need to restart your Rails application if you make any changes to the gem**.
> 
> ## Test NPM for react-on-rails
> ```

Testing the Node package for react-on-rails

---

*[CONTRIBUTING.md, line 95 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIBHXzxKMvwK6Bd9C7:-KeIBHXzxKMvwK6Bd9C8:b-89zspa) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/CONTRIBUTING.md#L95)):*
> You can also test the npm parts of the gem with an external application.

In addition to testing the Ruby parts out, you can test the node package parts of the gem with an external application.

---

*[CONTRIBUTING.md, line 107 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIBZ-Oy_Y2DfLTyUzz:-KeIBZ-PjrbPDJvrvIGC:b-7m9qis) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/CONTRIBUTING.md#L107)):*
> ```Markdown
> cd test/client
> yarn remove react-on-rails
> ```
> ```

was this necessary? did you try just running the npm command?

---

*[CONTRIBUTING.md, line 114 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIBdZeyBU17VKshmQb:-KeIBdZeyBU17VKshmQc:bqihn6s) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/CONTRIBUTING.md#L114)):*
> ```Markdown
> npm i --save 'react-on-rails@../path-to-react-on-rails'
> ```
> _Note: You must use npm here till yarn stops preferring cached packages over local. see [issue](https://github.com/yarnpkg/yarn/issues/2649)_
> ```

excellent! but put in issue number!

```
_Note: You must use npm here till yarn stops preferring cached packages over local. see [yarn issue 2649](https://github.com/yarnpkg/yarn/issues/2649)_
```

---

*[spec/dummy/client/package.json, line 36 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIC-vtQHB0gEuF6VCT:-KeIC-vtQHB0gEuF6VCU:b-cj6q1e) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/spec/dummy/client/package.json#L36)):*
> ```JSON
>     "react": "^15.4.1",
>     "react-dom": "^15.4.1",
>     "react-on-rails": "file:../../../../react_on_rails",
> ```

Is this necessary to put in react_on_rails? what if the project is cloned under a different name?

less dots and no react_on_rails maybe!

---

*[spec/dummy/client/package.json, line 77 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIBvwIyWfC1IaRj8RR:-KeIBvwIyWfC1IaRj8RS:b-5l76x7) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/spec/dummy/client/package.json#L77)):*
> ```JSON
>     "build:server": "webpack --config webpack.server.rails.build.config.js",
>     "hot-assets": "babel-node server-rails-hot.js",
>     "install-react-on-rails": "npm install --save 'react-on-rails@file:../../../../react_on_rails'"
> ```

If you really need to delete, then you can add this to the script here.

Is this necessary to put in react_on_rails? what if the project is cloned under a different name?

less dots and no react_on_rails maybe!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-:-KeICNO7ETEM8tOpqLts:b14c7sg)*
<!-- Sent from Reviewable.io -->

justin808 commented:
@wanyamaman Please let me know when you've addressed my comments!

justin808 commented:
@wanyamaman Is this ready for merging?

wanyamaman commented:
@justin808 updated

---

Review status: 0 of 2 files reviewed at latest revision, 7 unresolved discussions.

---

*[CONTRIBUTING.md, line 85 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIB7Jjdjj0LM7_z6n1:-Kedgs13Oh2ZfGX8Za_x:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/CONTRIBUTING.md#L85)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Testing the Ruby Gem
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 94 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIBBO7X70btnvBYnV8:-Kedgsq4wyIqitNgqxUW:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/CONTRIBUTING.md#L94)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Testing the Node package for react-on-rails
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 95 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIBHXzxKMvwK6Bd9C7:-Kedgu6z7cf1Gs7PUKdr:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/CONTRIBUTING.md#L95)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

> You can also test the npm parts of the gem with an external application.

In addition to testing the Ruby parts out, you can test the node package parts of the gem with an external application.
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 107 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIBZ-Oy_Y2DfLTyUzz:-Ked5xzpRKZNQZlxa9ju:bcneuzt) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/CONTRIBUTING.md#L107)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

was this necessary? did you try just running the npm command?
</blockquote></details>

It's not necessary, this was used to clean the `yarn.lock` file for those who've already ran the install using yarn instead of npm.

---

*[CONTRIBUTING.md, line 114 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIBdZeyBU17VKshmQb:-KedgvDAz4zNbnhTMnTs:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/CONTRIBUTING.md#L114)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

excellent! but put in issue number!

```
_Note: You must use npm here till yarn stops preferring cached packages over local. see [yarn issue 2649](https://github.com/yarnpkg/yarn/issues/2649)_
```
</blockquote></details>

Done.

---

*[spec/dummy/client/package.json, line 36 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIC-vtQHB0gEuF6VCT:-Kedgngah7FtCwURrjvH:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/spec/dummy/client/package.json#L36)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Is this necessary to put in react_on_rails? what if the project is cloned under a different name?

less dots and no react_on_rails maybe!
</blockquote></details>

Done.

---

*[spec/dummy/client/package.json, line 77 at r4](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KeIBvwIyWfC1IaRj8RR:-KedgqGcZ-izCiwyYAQu:b-896fix) ([raw file](https://github.com/shakacode/react_on_rails/blob/82754ceb5efda1fd22efaf701943c2e857307fd5/spec/dummy/client/package.json#L77)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

If you really need to delete, then you can add this to the script here.

Is this necessary to put in react_on_rails? what if the project is cloned under a different name?

less dots and no react_on_rails maybe!
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-:-Kedgwprxf57G5WzP1OK:beycucv)*
<!-- Sent from Reviewable.io -->

justin808 commented:
One question...couple minor comments!

Close!

---

Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[CONTRIBUTING.md, line 113 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-Kefy0k9AqMleu5nBJYM:-Kefy0k9AqMleu5nBJYN:b49nm0q) ([raw file](https://github.com/shakacode/react_on_rails/blob/50d10b6caf6aa9bfa618f384498e5e4137be3cb5/CONTRIBUTING.md#L113)):*
> ```Markdown
> 
> #### Example: Testing NPM changes with the dummy app
> 1. Add `console.log('Hello!')` to top of the `clientStartup` function in `react_on_rails/node_package/src/clientStartup.js`. This is our desired change.
> ```

file, not function

---

*[CONTRIBUTING.md, line 115 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KefxlHkusbZ2RrDfaJZ:-KefxlHkusbZ2RrDfaJ_:b-hnpijd) ([raw file](https://github.com/shakacode/react_on_rails/blob/50d10b6caf6aa9bfa618f384498e5e4137be3cb5/CONTRIBUTING.md#L115)):*
> ```Markdown
> 1. Add `console.log('Hello!')` to top of the `clientStartup` function in `react_on_rails/node_package/src/clientStartup.js`. This is our desired change.
> 2. Run `yarn` and `yarn run build` from the gem root directory: `react_on_rails/`. This compiles our changes.
> 3. Run the install script `npm run install-react-on-rails` in `react_on_rails/spec/dummy` to copy over our changes to the dummy app. Alternatively, you can run `npm i --save 'file:../../../'` in `react_on_rails/spec/dummy/client`. Our NPM changes are now available in the dummy app.
> ```

Are we sure that the `npm i` does not run the build command? I'm pretty sure it used to. We should check if we can remove step 2.

---

*[CONTRIBUTING.md, line 116 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KefxtjNUrDZ_1aTvV6n:-KefxtjNUrDZ_1aTvV6o:b-ox1a9y) ([raw file](https://github.com/shakacode/react_on_rails/blob/50d10b6caf6aa9bfa618f384498e5e4137be3cb5/CONTRIBUTING.md#L116)):*
> ```Markdown
> 2. Run `yarn` and `yarn run build` from the gem root directory: `react_on_rails/`. This compiles our changes.
> 3. Run the install script `npm run install-react-on-rails` in `react_on_rails/spec/dummy` to copy over our changes to the dummy app. Alternatively, you can run `npm i --save 'file:../../../'` in `react_on_rails/spec/dummy/client`. Our NPM changes are now available in the dummy app.
> 4. Refresh the browser if the server is already running or start the server using `foreman start` from `react_on_rails/spec/dummy` and navigate to `http://localhost:3000/`. You will now see the `[SERVER] Hello!` message printed in the browser's console.
> ```

that's is only if you have server rendering turned on

and you modified the clientStartup file

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-:-KefyFc2t7tWg70JcLQw:b-j355cc)*
<!-- Sent from Reviewable.io -->

wanyamaman commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[CONTRIBUTING.md, line 115 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KefxlHkusbZ2RrDfaJZ:-KegtNtZ0x01NR_-abM4:bwbgx60) ([raw file](https://github.com/shakacode/react_on_rails/blob/50d10b6caf6aa9bfa618f384498e5e4137be3cb5/CONTRIBUTING.md#L115)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

Are we sure that the `npm i` does not run the build command? I'm pretty sure it used to. We should check if we can remove step 2.
</blockquote></details>

It does but yarn doesn't. Should we cater for yarn setups or remove that step and the yarn build instruction above until yarn fixes their caching issue?

---

*[CONTRIBUTING.md, line 116 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KefxtjNUrDZ_1aTvV6n:-Kegu2JkAWn-Epk71mMM:bbighi5) ([raw file](https://github.com/shakacode/react_on_rails/blob/50d10b6caf6aa9bfa618f384498e5e4137be3cb5/CONTRIBUTING.md#L116)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

that's is only if you have server rendering turned on

and you modified the clientStartup file
</blockquote></details>

Should that be added as a footnote? The assumption here is that you haven't modified spec/dummy so server rendering is on and you've modified the `clientStartup` file as described. 

The whole point of this section is to give the reader a general reproducible example highlighting the flow of execution when making a change to the npm package. Unless you'd like to cover all possible configurations they should be able to deduce the changes in their specific configuration. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737)*
<!-- Sent from Reviewable.io -->

justin808 commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[CONTRIBUTING.md, line 115 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KefxlHkusbZ2RrDfaJZ:-KelRY3uU3hdVGYyszc6:b6708v6) ([raw file](https://github.com/shakacode/react_on_rails/blob/50d10b6caf6aa9bfa618f384498e5e4137be3cb5/CONTRIBUTING.md#L115)):*
<details><summary><i>Previously, wanyamaman (William Wanyama) wrote…</i></summary><blockquote>

It does but yarn doesn't. Should we cater for yarn setups or remove that step and the yarn build instruction above until yarn fixes their caching issue?
</blockquote></details>

I'd mention that, so we know that when yarn fixes that, we'll remove that.

---

*[CONTRIBUTING.md, line 116 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KefxtjNUrDZ_1aTvV6n:-KelRcLPRV-MsQuz9nTm:b-zge8hc) ([raw file](https://github.com/shakacode/react_on_rails/blob/50d10b6caf6aa9bfa618f384498e5e4137be3cb5/CONTRIBUTING.md#L116)):*
<details><summary><i>Previously, wanyamaman (William Wanyama) wrote…</i></summary><blockquote>

Should that be added as a footnote? The assumption here is that you haven't modified spec/dummy so server rendering is on and you've modified the `clientStartup` file as described. 

The whole point of this section is to give the reader a general reproducible example highlighting the flow of execution when making a change to the npm package. Unless you'd like to cover all possible configurations they should be able to deduce the changes in their specific configuration. 
</blockquote></details>

I'd keep this all client side, and if the client side works, we'll know the server side works.

so your example with `[SERVER]` is not right.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737)*
<!-- Sent from Reviewable.io -->

justin808 commented:
@wanyamaman Would you like to address my last few comments and we'll merge?

wanyamaman commented:
Addressed

---

Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

*[CONTRIBUTING.md, line 113 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-Kefy0k9AqMleu5nBJYM:-Kf8kTieoYP-fqKh3dU-:bvnrpcm) ([raw file](https://github.com/shakacode/react_on_rails/blob/50d10b6caf6aa9bfa618f384498e5e4137be3cb5/CONTRIBUTING.md#L113)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

file, not function
</blockquote></details>

updated

---

*[CONTRIBUTING.md, line 115 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KefxlHkusbZ2RrDfaJZ:-Kf8l0LESC-VFRbbu1PL:b-i460fr) ([raw file](https://github.com/shakacode/react_on_rails/blob/50d10b6caf6aa9bfa618f384498e5e4137be3cb5/CONTRIBUTING.md#L115)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

I'd mention that, so we know that when yarn fixes that, we'll remove that.
</blockquote></details>

I've removed it and left a footnote on how to do it with yarn once the issue is fixed.

---

*[CONTRIBUTING.md, line 116 at r5](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-KefxtjNUrDZ_1aTvV6n:-Kf8k_OoFpRnNCb-TuwM:b-nihyvf) ([raw file](https://github.com/shakacode/react_on_rails/blob/50d10b6caf6aa9bfa618f384498e5e4137be3cb5/CONTRIBUTING.md#L116)):*
<details><summary><i>Previously, justin808 (Justin Gordon) wrote…</i></summary><blockquote>

I'd keep this all client side, and if the client side works, we'll know the server side works.

so your example with `[SERVER]` is not right.
</blockquote></details>

Moved to client side.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-:-Kf8l1wkDh4-rORR6ulG:bx3iq6m)*
<!-- Sent from Reviewable.io -->

justin808 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[CONTRIBUTING.md, line 113 at r6](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-Kf9ZdYedHN6ViMR1f4F:-Kf9ZdYedHN6ViMR1f4G:b5v33ft) ([raw file](https://github.com/shakacode/react_on_rails/blob/444f048ae04fa9ab483de5beacebb25a7f2a86d9/CONTRIBUTING.md#L113)):*
> 1. Add `console.log('Hello!')` [here](https://github.com/shakacode/react_on_rails/blob/master/node_package/src/clientStartup.js#L181) in `react_on_rails/node_package/src/clientStartup.js`. This is our desired change.

```
1. Add `console.log('Hello!')` [here](https://github.com/shakacode/react_on_rails/blob/master/node_package/src/clientStartup.js#L181) in `react_on_rails/node_package/src/clientStartup.js` to confirm we're getting an update to the node package.
```

I can do this update.

---

*[CONTRIBUTING.md, line 115 at r6](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-Kf9ZrmTLlj5NM7DCmUW:-Kf9ZrmTLlj5NM7DCmUX:b-qs5zhk) ([raw file](https://github.com/shakacode/react_on_rails/blob/444f048ae04fa9ab483de5beacebb25a7f2a86d9/CONTRIBUTING.md#L115)):*
> ```Markdown
> 1. Add `console.log('Hello!')` [here](https://github.com/shakacode/react_on_rails/blob/master/node_package/src/clientStartup.js#L181) in `react_on_rails/node_package/src/clientStartup.js`. This is our desired change.
> 2. Run the install script `npm run install-react-on-rails` in `react_on_rails/spec/dummy` to copy over our changes to the dummy app. Alternatively, you can run `npm i --save 'file:../../../'` in `react_on_rails/spec/dummy/client`. Our NPM changes are now available in the dummy app.
> 3. Refresh the browser if the server is already running or start the server using `foreman start` from `react_on_rails/spec/dummy` and navigate to `http://localhost:5000/`. You will now see the `Hello!` message printed in the browser's console.
> ```

using `foreman start -f Profile.dev`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/737#-:-KfKZwQkNMr7uVQoRY_v:bnfp4nl)*
<!-- Sent from Reviewable.io -->
